dataset/instrution_commit_clean.py

```diff
@@ -0,0 +1,312 @@
+    if lower_subject.startswith(" ") and len(lower_subject.strip().split()) == 1:
+        return False
+
+    return True
+
+
+def check_uniques(example, uniques):
+    """Check if current hash is still in set of unique hashes and remove if true."""
+    if example["repos"] + "/" + example["old_file"] in uniques:
+        uniques.remove(example["repos"] + "/" + example["old_file"])
+        return True
+    else:
+        return False
+
+
+uniques = set([r + "/" + filename for filename, r in zip(ds["old_file"], ds["repos"])])
+print(len(uniques))
+
+ds = ds.filter(check_uniques, fn_kwargs={"uniques": uniques}, num_proc=1)
+
+ds_clean = ds.filter(commit_filter, num_proc=30)
+
+print("After commit filtering, the dataset size is {}".format(len(ds_clean)))
+
+
+def prepare_code(example):
+    if np.random.random() < FULL_RANGE_FRAC:
+        example["content"] = f"<commit_before>{example['old_contents']}<commit_msg>{example['subject']}<commit_after>{example['new_contents']}"
+        example["size"] = len(example["content"])
+    else:
+        start_offset = np.random.randint(MIN_RANGE, MAX_RANGE)
+        end_offset = np.random.randint(MIN_RANGE, MAX_RANGE)
+
+        old_lines = example["old_contents"].splitlines()
+        new_lines = example["new_contents"].splitlines()
+
+        old_start = max(0, example["old_change_start"] - start_offset)
+        new_start = max(0, example["new_change_start"] - start_offset)
+
+        old_end = min(len(old_lines), example["old_change_end"] + end_offset)
+        new_end = min(len(new_lines), example["new_change_end"] + end_offset)
+
+        code_before = "\n".join(old_lines[old_start:old_end])
+        code_after = "\n".join(new_lines[new_start:new_end])
+        example["content"] = f"<commit_before>{code_before}<commit_msg>{example['subject']}<commit_after>{code_after}"
+        example["size"] = len(example["content"])
+    return example
+
+
+ds_clean = ds_clean.map(prepare_code, num_proc=30)
+opt_out_github_login_the_stack_12 = set([el.lower() for el in opt_out_github_login_the_stack_12])
+
+
+def filter_opt_out(example):
+    repo_names = example["repos"].split(",")
+    try:
+        repo_owners = [repo_name.split("/")[0].lower() for repo_name in repo_names]
+    except:
+        print(example["repos"])
+    filtered_repo_names = [repo_name for repo_name, repo_owner in zip(repo_names,
+                                                                      repo_owners) if
+                           repo_owner not in opt_out_github_login_the_stack_12]
+
+    if len(filtered_repo_names) > 0:
+        example["repos"] = ",".join(filtered_repo_names)
+        example["drop_opt_out"] = False
+    else:
+        example["repos"] = ""
+        example["drop_opt_out"] = True
+    return example
+
+
+ds_opt_out = ds_clean.map(filter_opt_out, num_proc=30)
+ds_opt_out = ds_opt_out.filter(lambda x: not x["drop_opt_out"], num_proc=30)
```

Comment:
The opt out filtering is already done on the uploaded dataset

Reply:
Great! Will remove it.